tools/clang/tools/dxclib/dxc.cpp

```diff
@@ -822,33 +822,38 @@ int DxcContext::MaybeRunExternalValidatorAndPrintValidationOutput(
                               !opts.OptDump && !opts.DumpDependencies &&
                               !opts.VerifyDiagnostics;
   bool needsValidation = produceFullContainer && !opts.DisableValidation;
-  if (needsValidation) {
+  if (!needsValidation) {
+    return S_OK;
+  }
 
-    CComPtr<IDxcValidator> pValidator;
-    IFT(CreateInstance(CLSID_DxcValidator, &pValidator));
+  CComPtr<IDxcValidator> pValidator;
+  IFT(CreateInstance(CLSID_DxcValidator, &pValidator));
 
-    CComPtr<IDxcBlob> pProgram;
-    CComPtr<IDxcOperationResult> pValResult;
+  CComPtr<IDxcBlob> pProgram;
+  CComPtr<IDxcOperationResult> pValResult;
 
-    IFT(pCompileResult->GetResult(&pProgram));
+  IFT(pCompileResult->GetResult(&pProgram));
 
-    IFT(pValidator->Validate(pProgram, DxcValidatorFlags_InPlaceEdit,
-                             &pValResult));
-    CComPtr<IDxcResult> pResult;
-    HRESULT ValHR;
-    pValResult->GetStatus(&ValHR);
-    if (DXC_FAILED(ValHR) && SUCCEEDED(pValResult->QueryInterface(&pResult))) {
-      CComPtr<IDxcBlobEncoding> pErrorBlob;
-      CComPtr<IDxcBlobWide> pErrorOutput;
+  IFT(pValidator->Validate(pProgram, DxcValidatorFlags_InPlaceEdit,
+                           &pValResult));
+  CComPtr<IDxcResult> pResult;
+  HRESULT ValHR;
+  pValResult->GetStatus(&ValHR);
 
-      IFT(pResult->GetOutput(DXC_OUT_ERRORS, IID_PPV_ARGS(&pErrorBlob),
-                             &pErrorOutput));
+  if (!DXC_FAILED(ValHR))
+    return ValHR;
```

Comment:
```suggestion
  if (SUCCEEDED(ValHR))
    return ValHR;
```

I would like to able to suggest:

```suggestion
  if (SUCCEEDED(ValHR))
    return S_OK;
```

But ... shudder ... you have to consider `S_FALSE` :(